xmtp/src/storage/encrypted_store/schema.rs

```diff
@@ -17,6 +17,16 @@ diesel::table! {
     }
 }
 
+diesel::table! {
+    installations (installation_id) {
+        installation_id -> Text,
+        user_address -> Text,
+        first_seen -> BigInt,
+        contact -> Binary,
+        expires_at -> Nullable<BigInt>,
+    }
```

Comment:
I think there is room to integrate both tables if desired.

> Would it makes sense to merge the installations table with the sessions

To integrate these session:vmac_session_data would need to change from Vec<u8> to Option<Vec<u8>> as session data will not exist when initialized. Doesn't seem terrible -- though every session will need to be validated tp ensure the session_data is present, rather than being type enforced.

> . In this case I think it makes sense to just use decryption failure/retry logic until both sides converge on the same session,

My understanding is that session reset is the solution to desynchronized states, which would create a new session. If the old session is immediately deleted once a reset has been initiated that seems like theres a high chance of message loss for inflight messages. I know decryption failure is OOS for now, but seems pertinent to think about wether historical (aka non active) sessions are retained for any period of time. If the answer is a flat no, then no issues here.

Viewing ContactBundles as a "Pre-Session" seems slightly odd to me as not all contact bundles result in a session, but I could make it happen. Thought I'd be particularly interested in how PreKey updates will work, seems like there will naively be multiple ContactBundles for an installation. If only one bundle is retained per user then on refresh clients will need to re-evaluate each other bundle to determine there are no new installations. Not critical at this stage though.

I've got no strong case against the proposal. Seems that One session per Installation could be achieved in other ways, but happy to implement it that way if preferred.